addons/web/static/tests/views/fields/numeric_fields_tests.js

```diff
@@ -180,4 +183,124 @@ QUnit.module("Fields", (hooks) => {
+            registry
+                .category("services")
+                .add(
+                    "localization",
+                    makeFakeLocalizationService({ decimalPoint: ",", thousandsSep: "." })
+                );
+
+            await makeView({
+                serverData,
+                type: "form",
+                resModel: "partner",
+                arch: /*xml*/ `
+                    <form>
+                        <field name="float_factor_field" options="{'factor': 0.5}"/>
+                        <field name="qux"/>
+                        <field name="int_field"/>
+                        <field name="monetary"/>
+                        <field name="currency_id" invisible="1"/>
+                        <field name="percentage"/>
+                        <field name="progressbar" widget="progressbar" options="{'editable': true, 'max_value': 'qux', 'edit_max_value': true, 'edit_current_value': true}"/>
+                    </form>`,
+                resId: 1,
+            });
+
+            // Get all inputs
+            const floatFactorField = target.querySelector(".o_field_float_factor input");
+            const floatInput = target.querySelector(".o_field_float input");
+            const integerInput = target.querySelector(".o_field_integer input");
+            const monetaryInput = target.querySelector(".o_field_monetary input");
+            const percentageInput = target.querySelector(".o_field_percentage input");
+
+            /**
+             * Common assertion steps are extracted in this procedure.
+             *
+             * @param {object} params
+             * @param {InputElement} params.el
+             * @param {[number, number]} params.selectionRange
+             * @param {string} params.expectedValue
+             * @param {string} params.msg
+             */
+            async function testInputElementOnNumpadDecimal(params) {
+                const { el, selectionRange, expectedValue, msg } = params;
+
+                el.focus();
+                el.setSelectionRange(...selectionRange);
+                el.dispatchEvent(new KeyboardEvent("keydown", { code: "NumpadDecimal", key: "." }));
+
+                // dispatch an extra keydown event and assert that it's not default prevented
+                const extraEvent = new KeyboardEvent("keydown", { code: "Digit1", key: "1" });
+                el.dispatchEvent(extraEvent);
+                assert.ok(
+                    !extraEvent.defaultPrevented,
+                    "Keydown event should not be default prevented"
+                );
+                // Selection range should be 1 + the specified selection start.
+                assert.strictEqual(el.selectionStart, selectionRange[0] + 1);
+                assert.strictEqual(el.selectionEnd, selectionRange[0] + 1);
+                await nextTick();
+                assert.strictEqual(el.value, expectedValue, msg || "");
+                click(document.body);
```

Comment:
what is it for ?

Reply:
It is for removing the focus to the current input. I can remove it because it's not really that important.